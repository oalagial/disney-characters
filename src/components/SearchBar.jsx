import { Button, TextField } from '@mui/material'
import React, { useState } from 'react'
import styled from 'styled-components'

const SearchBarContainer = styled.div`
  color: darkslategray;
  background-color: aliceblue;
  padding: 8px;
  border-radius: 4px;
  margin-bottom: 16px;
  display: flex;
  justify-content: space-between;
`

const SearchBar = ({ handleSearchButtonClicked }) => {
  const [filterTvShow, setFilterTvShow] = useState('')
  const [filterName, setFilterName] = useState('')

  return (
    <SearchBarContainer>
      <div style={{ width: '40%', display: 'flex' }}>
        <TextField
          label="Search by Name"
          data-testid="name-input"
          variant="outlined"
          onChange={(e) => {
            setFilterName(e.target.value)
          }}
          sx={{ flexGrow: 1, marginLeft: '10px' }}
        />
        <TextField
          label="Search by TV Show"
          data-testid="tvShow-input"
          variant="outlined"
          onChange={(e) => {
            setFilterTvShow(e.target.value)
          }}
          sx={{ flexGrow: 1, marginLeft: '10px' }}
        />
      </div>
      <Button
        variant="contained"
        onClick={() =>
          handleSearchButtonClicked({ name: filterName, tvShows: filterTvShow })
        }
      >
        Search
      </Button>
    </SearchBarContainer>
  )
}

export default SearchBar
