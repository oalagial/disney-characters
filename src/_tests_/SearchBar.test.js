import { render, screen } from '@testing-library/react'
import SearchBar from '../components/SearchBar'
import renderer from 'react-test-renderer'

describe('SearchBar', () => {
  test('renders correctly', () => {
    render(<SearchBar />)
    const inputName = screen.getByLabelText(/Search by name/i)
    const inputTvShow = screen.getByLabelText(/Search by TV Show/i)
    const submitButtonElement = screen.getByRole('button')

    expect(inputName).toBeInTheDocument()
    expect(inputTvShow).toBeInTheDocument()
    expect(submitButtonElement).toBeInTheDocument()
  })

  it('Matches DOM Snapshot', () => {
    const domTree = renderer.create(<SearchBar />).toJSON()
    expect(domTree).toMatchSnapshot()
  })
})
