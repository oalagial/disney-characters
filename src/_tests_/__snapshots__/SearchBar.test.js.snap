// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Matches DOM Snapshot 1`] = `
<div
  className="sc-bcXHqe bhDrEg"
>
  <div
    style={
      Object {
        "display": "flex",
        "width": "40%",
      }
    }
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-enattd-MuiFormControl-root-MuiTextField-root"
      data-testid="name-input"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Search by Name
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Search by Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root css-enattd-MuiFormControl-root-MuiTextField-root"
      data-testid="tvShow-input"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r1:"
        id=":r1:-label"
      >
        Search by TV Show
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r1:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Search by TV Show
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Search
  </button>
</div>
`;
